utils/cloneByDate.py

```diff
@@ -868,7 +879,7 @@ def __login(self):
     def list_channel_labels(self):
         self.auth_check()
         key = "chan_labels"
-        if self.cache.has_key(key):
+        if key in self.cache:
```

Comment:
Be careful here, there are some RHN classes that implements `has_key()` method and `2to3` is wrongly suggesting changes to `if x in y`. You need to check whether `salt.cache` is really a dict or some other class instance.

Reply:
@meaksh it seems it is a dict defined here https://github.com/uyuni-project/uyuni/blob/11da5a723d305016cc73d6519fe79b45028717bb/utils/cloneByDate.py#L854

Reply:
@meaksh i've check the other `has_key` fixes and the objects are dictionaries. Resolving.